spec/rspec/rails/matchers/have_enqueued_mail_spec.rb

```diff
@@ -393,22 +393,18 @@ def self.name; "NonMailerJob"; end
         }.to have_enqueued_mail(UnifiedMailer, :test_email).and have_enqueued_mail(UnifiedMailer, :email_with_args)
       end
 
-      it "matches arguments when mailer has only args" do
-        expect {
-          UnifiedMailer.email_with_args(1, 2).deliver_later
-        }.to have_enqueued_mail(UnifiedMailer, :email_with_args).with(1, 2)
-      end
-
-      it "matches arguments when mailer is parameterized" do
+      it "passes with provided argument matchers" do
         expect {
           UnifiedMailer.with('foo' => 'bar').test_email.deliver_later
-        }.to have_enqueued_mail(UnifiedMailer, :test_email).with('foo' => 'bar')
-      end
+        }.to have_enqueued_mail(UnifiedMailer, :test_email).with(
+          a_hash_including(params: {'foo' => 'bar'})
+        )
 
-      it "matches arguments when mixing parameterized and non-parameterized emails" do
         expect {
           UnifiedMailer.with('foo' => 'bar').email_with_args(1, 2).deliver_later
-        }.to have_enqueued_mail(UnifiedMailer, :email_with_args).with({'foo' => 'bar'}, 1, 2)
+        }.to have_enqueued_mail(UnifiedMailer, :email_with_args).with(
+          a_hash_including(params: {'foo' => 'bar'}, args: [1, 2])
+        )
```

Comment:
This just unpicks the changes to the behaviour

Reply:
I wonder if both:
```ruby
have_enqueued_mail(UnifiedMailer, :email_with_args).with(
          a_hash_including(params: {'foo' => 'bar'}, args: [1, 2])
        )
```
and 
```ruby
have_enqueued_mail(UnifiedMailer, :email_with_args).with({'foo' => 'bar'}, 1, 2)
```
would match.
I was believing that the change from #2516 made both match, but it's not confirmed in specs.